saleor_ai/management/commands/sync_db_ai.py

```diff
@@ -0,0 +1,120 @@
+import logging
+from collections import OrderedDict
+
+from django.core.management.base import BaseCommand
+
+from saleor.graphql.core.mutations import BaseMutation
+from saleor.product.models import ProductVariantChannelListing
+from saleor.product_class.models import ProductClassRecommendation
+from saleor.product_max_min.models import ProductMaxMin
+from saleor_ai.models import SaleorAI
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Sync product class, product max min in Saleor AI to DB."
+    chunk_size = 500
+
+    def update_metadata(self):
+        qs_product_classes = (
+            ProductClassRecommendation.objects.qs_filter_current_previous_one_query()
+        )
+        qs_products_max_min = (
+            ProductMaxMin.objects.qs_filter_current_previous_one_query()
+        )
+        for item in qs_product_classes:
+            listing = ProductVariantChannelListing.objects.get(id=item["listing_id"])
+            product_class_metadata = {
+                "product_class": {
+                    "current": item["current_version"],
+                    "previous": item["previous_version"],
+                }
+            }
+            listing.store_value_in_metadata(product_class_metadata)
+            listing.save()
+        for item in qs_products_max_min:
+            listing = ProductVariantChannelListing.objects.get(id=item["listing_id"])
+            product_class_metadata = {
+                "product_max_min": {
+                    "current": item["current_version"],
+                    "previous": item["previous_version"],
+                }
+            }
+            listing.store_value_in_metadata(product_class_metadata)
+            listing.save()
+
+    def handle(self, *args, **options):
+        group_saleor_ai = OrderedDict()
+        base_mutation = BaseMutation()
+        data_product_class = []
+        data_max_min = []
+        saleor_ai = SaleorAI.objects.values(
+            "article_code",
+            "franchise_code",
+            "min_qty",
+            "max_qty",
+            "start_of_week",
+            "product_class_qty",
+            "product_class_value",
+            "product_class_default",
+        ).all()
+        for item in saleor_ai:
```

Comment:
không nên làm thế này, nếu db to thì command em chạy đến bao giờ mới xong

em cần phải có ít nhất 1 cái mốc thời gian
hoặc là do user input
hoặc do em set giá trị mặc định

ngoài ra, user cũng có thể chỉ muốn sync data cho 1 số đối tượng nhất định, nên thêm tham số để user có thể truyền vào